### PR TITLE
fix: Skip calling ElementMapFactory::withInstance() if the instance is not set

### PR DESCRIPTION
**Pull request by hectoras**
**Associated Jira ticket:** [ADF-841](https://oat-sa.atlassian.net/browse/ADF-841)

This fixes calls being made when creating new instances from the remote publishing admin panel, as there is no instance then (and `null` is passed instead). `ElementMapFactory::withInstance()` will fail in such cases, as it expects a non-null parameter.

### CODE REVIEW

File: actions/form/class.Instance.php [hunk 1]
```diff
@@ -112,10 +112,13 @@ protected function initElements()
...
 
-            //map properties widgets to form elments
-            $element = $this->getElementFactory()
-                ->withInstance($instance)
-                ->create($property);
+            //map properties widgets to form elements
+            $elementFactory = $this->getElementFactory();
+            if($instance != null) {
+                $elementFactory->withInstance($instance);
+            }
```
Comment (gabrielfs7) [inline actions/form/class.Instance.php]: PSR and strict check :) 

```suggestion
            $elementFactory = $this->getElementFactory();
            
            if ($instance instanceof core_kernel_classes_Resource) {
                $elementFactory->withInstance($instance);
            }
```

Comment (hectoras) [reply actions/form/class.Instance.php]: Realized that after pushing :sweat: Fixed in a4d454316c584f602b3b5e65f5e4200112a59a7f

As you suggested, I'm  changing the `null` comparison with `instanceof` as well in d410e8f624d05ff2740c191857a4c033af6e2783

File: actions/form/class.Instance.php [hunk 2]
```diff
@@ -112,10 +110,13 @@ protected function initElements()
                 continue;
             }
 
-            //map properties widgets to form elments
-            $element = $this->getElementFactory()
-                ->withInstance($instance)
-                ->create($property);
+            //map properties widgets to form elements
+            $elementFactory = $this->getElementFactory();
+            if ($instance instanceof core_kernel_classes_Resource) {
```
Comment (shpran) [inline actions/form/class.Instance.php:115]:
```suggestion
            $elementFactory = $this->getElementFactory();

            if ($instance instanceof core_kernel_classes_Resource) {
```

**Review by shpran — APPROVED** [review]

File: actions/form/class.Instance.php [hunk 3]
```diff
@@ -112,10 +110,13 @@ protected function initElements()
                 continue;
             }
 
-            //map properties widgets to form elments
-            $element = $this->getElementFactory()
-                ->withInstance($instance)
-                ->create($property);
+            //map properties widgets to form elements
+            $elementFactory = $this->getElementFactory();
```
Comment (gabrielfs7) [inline actions/form/class.Instance.php:114]: Please do not forget you also need to fix it here https://github.com/oat-sa/tao-core/blob/develop/helpers/form/ElementMapFactory.php#L222

**Review by gabrielfs7 — CHANGES_REQUESTED** [review]
Needs to fix here too https://github.com/oat-sa/tao-core/blob/develop/helpers/form/ElementMapFactory.php#L222

EDIT: Fixed in a918ca61be521f9c9f4ebd300de99cf6d29f0ec9

File: actions/form/class.Instance.php (hunk 3 above)
Comment (hectoras) [reply actions/form/class.Instance.php:114]: Fixed in a918ca61be521f9c9f4ebd300de99cf6d29f0ec9 & b4458c0fcc774117f919d1128089cf0784d75843

File: actions/form/class.Instance.php (hunk 2 above)
Comment (hectoras) [reply actions/form/class.Instance.php]: Added separation between blocks in 8854ea2be826e7e1ec71c4917c8047727ad10f2f  :+1:

File: actions/form/class.Instance.php [hunk 4]
```diff
@@ -176,7 +178,7 @@ protected function initElements()
             $this->form->addElement($element[0]);
         }
 
-        //add an hidden elt for the class uri
+        // Add a hidden element for the class uri
```
[the remark this answers is not in the document]
Comment (hectoras) [reply actions/form/class.Instance.php]: Reverted 46afb28f82948efc58fb7d05307f5e52729bae28 :+1:

**Review by gabrielfs7 — APPROVED** [review]
Good job! 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by hectoras** [comment]
QA has verified this fix works as expected. Therefore, merging this fix back to the `develop` branch.

**Pull request merged**